InaccessibleObjectException when creating Glyph

When JavaFX is on the module-path, simply creating a `Glyph` throws the following exception:

```none
Exception in thread "main" java.lang.reflect.InvocationTargetException
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:564)
        at java.base/sun.launcher.LauncherHelper$FXHelper.main(LauncherHelper.java:1051)
Caused by: java.lang.RuntimeException: Exception in Application start method
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.launchApplication1(LauncherImpl.java:900)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication$2(LauncherImpl.java:195)
        at java.base/java.lang.Thread.run(Thread.java:832)
Caused by: java.util.ServiceConfigurationError: org.controlsfx.glyphfont.GlyphFont: Provider org.controlsfx.glyphfont.FontAwesome could not be instantiated
        at java.base/java.util.ServiceLoader.fail(ServiceLoader.java:584)
        at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:806)
        at java.base/java.util.ServiceLoader$ProviderImpl.get(ServiceLoader.java:724)
        at java.base/java.util.ServiceLoader$3.next(ServiceLoader.java:1396)
        at org.controlsfx.controls/org.controlsfx.glyphfont.GlyphFontRegistry.<clinit>(GlyphFontRegistry.java:67)
        at org.controlsfx.controls/org.controlsfx.glyphfont.Glyph.setFontFamily(Glyph.java:250)
        at org.controlsfx.controls/org.controlsfx.glyphfont.Glyph.<init>(Glyph.java:134)
        at org.controlsfx.controls/org.controlsfx.glyphfont.Glyph.create(Glyph.java:83)
        at app/com.example.app.Main.start(Main.java:13)
        at javafx.graphics/com.sun.javafx.application.LauncherImpl.lambda$launchApplication1$9(LauncherImpl.java:846)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runAndWait$12(PlatformImpl.java:455)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$10(PlatformImpl.java:428)
        at java.base/java.security.AccessController.doPrivileged(AccessController.java:391)
        at javafx.graphics/com.sun.javafx.application.PlatformImpl.lambda$runLater$11(PlatformImpl.java:427)
        at javafx.graphics/com.sun.glass.ui.InvokeLaterDispatcher$Future.run(InvokeLaterDispatcher.java:96)
        at javafx.graphics/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
        at javafx.graphics/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(WinApplication.java:174)
        ... 1 more
Caused by: java.lang.ExceptionInInitializerError
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
        at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
        at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
        at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
        at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
        at java.base/java.util.ServiceLoader$ProviderImpl.newInstance(ServiceLoader.java:782)
        ... 16 more
Caused by: java.lang.reflect.InaccessibleObjectException: Unable to make public static com.sun.javafx.css.StyleManager com.sun.javafx.css.StyleManager.getInstance() accessible: module javafx.graphics does not "exports com.sun.javafx.css" to module org.controlsfx.controls
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:349)
        at java.base/java.lang.reflect.AccessibleObject.checkCanSetAccessible(AccessibleObject.java:289)
        at java.base/java.lang.reflect.Method.checkCanSetAccessible(Method.java:196)
        at java.base/java.lang.reflect.Method.setAccessible(Method.java:190)
        at org.controlsfx.controls/impl.org.controlsfx.ReflectionUtils.addUserAgentStylesheet(ReflectionUtils.java:225)
        at org.controlsfx.controls/org.controlsfx.glyphfont.GlyphFont.<clinit>(GlyphFont.java:57)
        ... 22 more
```

---

## Example Code

module-info.java:

```java
module app {
  requires org.controlsfx.controls;

  exports com.example.app to
      javafx.graphics;
}
```

Main.java:

```java
package com.example.app;

import javafx.application.Application;
import javafx.scene.Scene;
import javafx.scene.layout.StackPane;
import javafx.stage.Stage;
import org.controlsfx.glyphfont.Glyph;

public class Main extends Application {

  @Override
  public void start(Stage primaryStage) {
    Glyph glyph = Glyph.create("FontAwesome|Gear"); // exception thrown here
    primaryStage.setScene(new Scene(new StackPane(glyph), 300, 150));
    primaryStage.show();
  }
}
```

Command line:

```none
java -p $mods -m app/com.example.app.Main
```

Where `$mods` is a path containing the `javafx.base`, `javafx.graphics`, `javafx.controls`, `org.controlsfx.controls`, and `app` modules. All except the `app` module (which must be compiled from the above code) were taken from Maven Central.

### Versions

Library         |    Version
 :---:             |   :---:
Java             | AdoptOpenJDK 14
JavaFX         | OpenJFX 14
ControlsFX  | 11.0.1

### Workaround

Use:

```none
--add-exports javafx.graphics/com.sun.javafx.css=org.controlsfx.controls
```

--- a/build.gradle
+++ b/build.gradle
@@ -2,8 +2,6 @@ ext.java9Args = [
         // For various usages of TraversalEngine
         "--add-exports=javafx.graphics/com.sun.javafx.scene=org.controlsfx.controls",
         "--add-exports=javafx.graphics/com.sun.javafx.scene.traversal=org.controlsfx.controls",
-        // For FontAwesome Style
-        "--add-exports=javafx.graphics/com.sun.javafx.css=org.controlsfx.controls",
         // For various behaviors across controls
         "--add-exports=javafx.controls/com.sun.javafx.scene.control.behavior=org.controlsfx.controls",
         // For ReadOnlyUnbackedObservableList across files
@@ -16,8 +14,6 @@ ext.java9Args = [
         "--add-exports=javafx.base/com.sun.javafx.collections=org.controlsfx.controls",
         // For VersionInfo in VersionChecker
         "--add-exports=javafx.base/com.sun.javafx.runtime=org.controlsfx.controls",
-        // For enabling FontAwesome
-        "--add-exports=javafx.graphics/com.sun.javafx.css=org.controlsfx.controls"
 ]
 
 ext.java9RuntimeArgs = [

--- a/controlsfx/src/main/java/impl/org/controlsfx/ReflectionUtils.java
+++ b/controlsfx/src/main/java/impl/org/controlsfx/ReflectionUtils.java
@@ -238,26 +238,6 @@ public class ReflectionUtils {
 
     /****************************************************************************************************
      *
-     * StyleManager
-     *
-     ****************************************************************************************************/
-
-    public static void addUserAgentStylesheet(String stylesheet) {
-        try {
-            Class<?> styleManagerClass = Class.forName("com.sun.javafx.css.StyleManager");
-            Method getInstance = styleManagerClass.getMethod("getInstance");
-            getInstance.setAccessible(true);
-            Object styleManager = getInstance.invoke(styleManagerClass);
-            Method addUserStyleSheet = styleManagerClass.getMethod("addUserAgentStylesheet", String.class);
-            addUserStyleSheet.setAccessible(true);
-            addUserStyleSheet.invoke(styleManager, stylesheet);
-        } catch (IllegalAccessException | ClassNotFoundException | NoSuchMethodException | InvocationTargetException e) {
-            throw new RuntimeException("Cannot add UserAgentStylesheet as the method is not accessible");
-        }
-    }
-
-    /****************************************************************************************************
-     *
      * TraversalEngine
      *
      ****************************************************************************************************/

--- a/controlsfx/src/main/java/org/controlsfx/glyphfont/Glyph.java
+++ b/controlsfx/src/main/java/org/controlsfx/glyphfont/Glyph.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2013, 2015 ControlsFX
+ * Copyright (c) 2013, 2015, 2020 ControlsFX
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -233,6 +233,10 @@ public class Glyph extends Label implements Duplicatable<Glyph> {
         }}
                 .fontFamily(getFontFamily())
                 .size(getFontSize());
+    }
+
+    @Override public String getUserAgentStylesheet() {
+        return Glyph.class.getResource("glyphfont.css").toExternalForm();
     }
 
     /***************************************************************************

--- a/controlsfx/src/main/java/org/controlsfx/glyphfont/Glyph.java
+++ b/controlsfx/src/main/java/org/controlsfx/glyphfont/Glyph.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2013, 2015, 2020 ControlsFX
+ * Copyright (c) 2013, 2020 ControlsFX
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without

--- a/controlsfx/src/main/java/org/controlsfx/glyphfont/GlyphFont.java
+++ b/controlsfx/src/main/java/org/controlsfx/glyphfont/GlyphFont.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2013, 2015, 2020 ControlsFX
+ * Copyright (c) 2013, 2020 ControlsFX
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without

--- a/controlsfx/src/main/java/org/controlsfx/glyphfont/GlyphFont.java
+++ b/controlsfx/src/main/java/org/controlsfx/glyphfont/GlyphFont.java
@@ -1,5 +1,5 @@
 /**
- * Copyright (c) 2013, 2015, ControlsFX
+ * Copyright (c) 2013, 2015, 2020 ControlsFX
  * All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
@@ -26,7 +26,6 @@
  */
 package org.controlsfx.glyphfont;
 
-import impl.org.controlsfx.ReflectionUtils;
 import javafx.scene.text.Font;
 
 import java.io.InputStream;
@@ -52,11 +51,6 @@ import java.util.Map;
  * <center><img src="glyphFont.png" alt="Screenshot of GlyphFont"></center>
  */
 public class GlyphFont {
-
-    static {
-        ReflectionUtils.addUserAgentStylesheet(
-                GlyphFont.class.getResource("glyphfont.css").toExternalForm()); //$NON-NLS-1$
-    }
 
     /***************************************************************************
      *                                                                         *

